### DISCUSSION TITLE
Should the HTTP client be removed?

### DISCUSSION THREAD

**Discussion by RobertoPrevato** [Neoteroi/BlackSheep#106] [Ideas]
I implemented the HTTP Client in BlackSheep mostly because I considered the client to be a good component to ship together with the server part, and to ensure that the `Request` and `Response` implementations are sound.

However, the focus on this project is revolving around the web framework part, and if the client was removed, it would be possible to remove the `httptools` dependency.

Comments are welcome.

**Comment by kun1996** [comment]
Many server frameworks do not implement the client themselves, so that the framework does not interact well with the client. Although the interface provided by the client is very convenient.



Now many services invoke other services via HTTP, and I believe it would be easier for us to process the result of the response if we could interact with the framework.



Above all, we are **blacksheep**.
